src/handlers/google_callback_handler.erl

```diff
@@ -0,0 +1,86 @@
+-module(google_callback_handler).
+
+%%% REST handler that receives the authentication code set by Google in the OAuth callback.
+%%% If the user is new, register it with the data from Google
+%%% In any case, respond with an access_token
+
+-export([init/3,
+         allowed_methods/2,
+         content_types_accepted/2,
+         content_types_provided/2,
+         from_json/2]).
+
+init(_Transport, _Req, []) ->
+  {upgrade, protocol, cowboy_rest}.
+
+allowed_methods(Req, State) ->
+  {[<<"POST">>], Req, State}.
+
+content_types_accepted(Req, State) ->
+  {[{<<"application/json">>, from_json}], Req, State}.
+
+content_types_provided(Req, State) ->
+  {[{<<"application/json">>, to_json}], Req, State}.
+
+from_json(Req, State) ->
+  {ok, Body, Req2} = cowboy_req:body(Req),
+
+  #{code := Code} = hp_json:decode(Body),
+
+  GoogleToken = get_google_access_token(Code),
+  Profile = get_public_profile(GoogleToken),
+
+  Email = get_primary_field(Profile, emailAddresses),
+  Name = get_primary_field(Profile, names),
+  AvatarUrl = get_primary_field(Profile, photos),
+
+  ok = register_user(Email, Name),
+  Token = hp_auth:build_holiday_access_token(Email, Name, AvatarUrl),
+
+  Encoded = hp_json:encode(#{access_token => Token}),
+  Req3 = cowboy_req:set_resp_body(Encoded, Req2),
+  {true, Req3, State}.
+
+%% internal
+get_google_access_token(Code) ->
+  ClientId = list_to_binary(os:getenv("GOOGLE_CLIENTID", false)),
```

Comment:
You may want to add this to the [credentials section of the README](https://github.com/lambdaclass/holiday_pinger#credentials)